sdk/helper/storagepacker/types.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+option go_package = "github.com/hashicorp/vault/sdk/helper/storagepacker";
+
+package storagepacker;
+
+import "google/protobuf/any.proto";
+
+// Item represents an entry that gets inserted into the storage packer
+message Item {
+	// ID must be provided by the caller; the same value, if used with GetItem,
+	// can be used to fetch the item. However, when iterating through a bucket,
+	// this ID will be an internal ID. In other words, outside of the use-case
+	// described above, the caller *must not* rely on this value to be
+	// consistent with what they passed in.
+	string id = 1;
+	// Message holds the contents of the item
+	// Deprecated: Use 'Data' instead
+	google.protobuf.Any message = 2;
+	// Data holds the contents of the item. Used in storage packer v2.
+	bytes data = 3;
+}
+
+// Bucket is a construct to hold multiple items within itself. This
+// abstraction contains multiple buckets of the same kind within itself and
+// shares amont them the items that get inserted. When the bucket as a whole
+// gets too big to hold more items, the contained buckets gets pushed out only
+// to become independent buckets. Hence, this can grow infinitely in terms of
+// storage space for items that get inserted.
+message Bucket {
+	// Key is the storage path where the bucket gets stored
+	string key = 1;
+	// Items holds the items contained within this bucket. Used by v1.
+	repeated Item items = 2;
```

Comment:
Why is this here? This is a different lib altogether so it doesn't seem like we need this field.

Reply:
Why was it present in your version? Is there an upgrade use case that I need to know about?

I *think* the upgrade will just be reading using SPv1 code and writing using SPv2 code, but the RFC didn't address it and there seem to be bits and pieces here where you seemed to be hinting at something else.

Reply:
Removed all the unnecessary fields, and changed Item to a normal type instead of a protobuf type.